src/Model/NotificationModel.php

```diff
@@ -13,39 +13,133 @@
 
 class NotificationModel implements NotificationInterface
 {
-    private ?string $url = null;
+    private string $id;
 
-    public function __construct(private string $id, private string $message, private string $type = Constants::TYPE_INFO)
+    // Simple link
+    private ?string $message       = null;
+    private ?string $type          = null;
+    private ?string $url           = null;
+    private bool    $active        = false;
+    private bool    $disabled      = false;
+    private bool    $withBadge     = true;
+    private bool    $badgeAnimated = true;
+
+    // Custom HTML
+    private ?string $html = null;
+
+    public function isValid(): bool
     {
+        return $this->message || $this->html;
+    }
+
+    public function __construct(
+        string $id,
+        ?string $message = null,
+        ?string $type = Constants::TYPE_INFO
+    ) {
+        $this->id      = $id;
+        $this->message = $message;
+        $this->type    = $type;
     }
 
     public function getIdentifier(): string
     {
         return $this->id;
     }
 
-    public function getType(): string
+    public function getMessage(): ?string
     {
-        return $this->type;
+        return $this->message;
     }
 
-    public function setMessage(string $message): void
+    public function setMessage(?string $message): self
     {
         $this->message = $message;
+
+        return $this;
```

Comment:
👎 for fluent interfaces

Reply:
OK, can you explain me why?
(interface is only for getter not setter)

Reply:
Not agree on all he wrote down, but a lot of good points:
https://ocramius.github.io/blog/fluent-interfaces-are-evil/

But I don't think that we should do that, because the simple question is: what is the real benefit we win and what do we give up?

Reply:
Yeah, I don't agree either on some point.
OK with the readability, and not enough experiences on my side with Proxy classes.
But the rest of it is kinda "meh". 
Maybe if he does that article with the current PHP version (not PHP 5.5 - 2013).

--------------

Anyway, if you don't want to, we'll not to.
The benefit are:
1. From 
  ```php
    $notification = new NotificationModel('1', '...');
    $notification->setMessage("Oops wrong site");
    $notification->setUrl("www.google.com");
    $notification->setActive(true);
    $notification->setBadgeAnimated(false);
  ```
2. To
  ```php
   $notification = (new NotificationModel('1', '...'))
        ->setMessage("Oops wrong site")
        ->setUrl("www.google.com")
        ->setActive(true)
        ->setBadgeAnimated(false)
    ;
  ```
That's it.

Cons:
At our level (not a framework), I don't know 🤷‍♂️